cl/recap/tasks.py

```diff
@@ -1415,3 +1420,90 @@ def mark_fq_status(fq, msg, status):
     if status == PROCESSING_STATUS.SUCCESSFUL:
         fq.date_completed = now()
     fq.save()
+
+def mark_epq_status(epq, msg, status):
+    """Mark the processing queue item as some process, and log the message.
+
+    :param pq: The ProcessingQueue object to manipulate
+    :param msg: The message to log and to save to pq's error_message field.
+    :param status: A pq status code as defined on the ProcessingQueue model.
+    """
+    if msg:
+        logger.info(msg)
+    epq.status_message = msg
+    epq.status = status
+    epq.save()
+    return epq.status, epq.status_message
+
+def decode_from_base64(content: str) -> str:
+    base64_bytes = content.encode('ascii')
+    decoded_bytes = base64.b64decode(base64_bytes)
+    return decoded_bytes.decode('ascii')
+
+@app.task(
+    bind=True, max_retries=5, interval_start=5 * 60, interval_step=10 * 60
+)
+def process_recap_email(self: Task, pk: int) -> Optional[int]:
+    start_time = now()
+    epq = EmailProcessingQueue.objects.get(pk=pk)
+    mark_epq_status(epq, "", PROCESSING_STATUS.IN_PROGRESS)
+
+    try:
+        message = epq.filepath.read()
+    except IOError as exc:
+        msg = "Internal processing error (%s: %s)." % (exc.errno, exc.strerror)
+        if (self.request.retries == self.max_retries):
+            mark_epq_status(epq, msg, PROCESSING_STATUS.FAILED)
+            return None
+        else:
+            mark_epq_status(epq, msg, PROCESSING_STATUS.QUEUED_FOR_RETRY)
+            raise self.retry(exc=exc)
+
+    parsed = json.loads(message)
+
+    # NOTE - I need to do some research into this: https://docs.aws.amazon.com/ses/latest/DeveloperGuide/receiving-email-action-lambda-event.html
+    body = decode_from_base64(parsed.get("content"))
+
+    # report = NotificationEmail(map_cl_to_pacer_id(epq.court_id))
+    # report._parse_text(body)
+    #
+    # data = report.data
+    data= {}
+
+    if data == {}:
+        msg = "Not a valid notification email. No message content."
+        mark_epq_status(epq, msg, PROCESSING_STATUS.INVALID_CONTENT)
+        self.request.chain = None
+        return None
+
+    docket = find_docket_object(epq.court_id, data["pacer_case_id"], data["docket_number"])
+
+    docket.add_recap_source()
+
+    # QUESTION - Do we want to update_docket_metadata(...) here?
+
+    if not docket.pacer_case_id:
+        docket.pacer_case_id = data["pacer_case_id"]
+
+    docket.save()
+
+    # QUESTION - Do we need to save any PacerHtmlFiles here? I presume those would be the download pages.
+
+    rds_created, content_updated = add_docket_entries(
+        docket, data["docket_entries"]
+    )
+    process_orphan_documents(rds_created, epq.court_id, docket.date_filed)
```

Comment:
TLDR: There's vanishingly little need to check for orphaned docs when doing notification emails. You can remove this line.

-----

Orphan docs is an interesting one, but I don't think it applies here — mostly. The idea of this function is that sometimes we get PDFs before we get metadata about those PDFs, with the result being that we have a doc, we have its pacer_doc_id, but we don't know what damned docket is associated with that pacer_doc_id. So...we call it an orphan — it has no parents — and we set it aside in case a parent ever comes along. Someday, maybe, we get that docket, and it says, I have the following 50 documents. When that happens, we go and check our orphanage (oof, this metaphor), and we check if we have any orphaned files that can be added to the docket. Sometimes we'll get a docket and suddenly have a dozen or more docs we can add to it.

So, all of that said, generally when we deal with email notifications, we will only have docket information about that one document, _and_ that we will have the PDF for that document, so we won't have metadata about docs that we don't have the PDFs for. If that's the case, there's no need to track down orphans. 

BUT: Say we get an email notification, we go try to buy it, and the magic link fails. In that case, checking for orphaned PDFs would make sense. We'd have the metadata for a doc, but lack the doc. Maybe it's one we already have from another source? 

I have to say, this is a triple edge case: We'd have to get an orphaned doc, it'd have to happen *before* processing the PDF from a notification email, AND the magic link would have to fail. Now that I've typed all this out, I'm realizing we wouldn't need to check for orphan docs, really and this line can be removed.